SoundCrate/scfront
# [ Annesa ] User Modals

## Changes
- Updated settings modal and API route 
- Updated next-auth to allow client-side changes to the user data in session storage. This needed to be done so that changes apply.
- Moved files around so that they made more sense (SettingsModal and UpdateCredentialsModal were switched, moved API routes to /user directory, changed name from SettingsModal to UpdateUserModal)

## Testing
- Sign into a user account
- Open the 'My Account' option in the dropdown menu item in the navigation bar
- Paste an image url and click preview. Confirm that the image changes (if the image is a valid one, if not it should be the default user).
- Save changes and ensure that the image actually updates on the site and in the navbar. Navbar should update immediately but may need to refresh in order to update all user profile images on the page.
- Open the 'Settings' option in the dropdown menu and confirm the frontend looks correct

## Images
<img width="1451" alt="image" src="https://github.com/SoundCrate/scfront/assets/93303711/a55c0bc5-c96d-409b-8554-00b1799e700b">
<img width="1451" alt="image" src="https://github.com/SoundCrate/scfront/assets/93303711/81d93d1c-1233-47e8-b7eb-48315187cf37">

# Todo
- The UpdateCredentialsModal is frontend ONLY, it's not connected to the backend at all. It also doesn't use any validation for password/email
- Still need to add validation to username for the UpdateUserModal. Front-end restricts username to < 30 characters but the username isn't getting validated in the backend at all yet.

## Todo (Nit)
- UI doesn't handle a broken image URL link very well
- I haven't fully tested the ability to change the username since doing so on the janedoe branch caused issues with the old lists backend. I've checked that it updates the database, but I haven't been able to test the frontend updates immediately after changing the username.
- The profile image in navbar remains highlighted while modal is open.
